payjoin/src/uri.rs

```diff
@@ -274,10 +312,103 @@ enum InternalPjParseError {
     UnsecureEndpoint,
 }
 
+/// Builder for `bip21::Uri` with `PayjoinParams`.
+///
+/// `amount` parameter is optional and can be changed with `[PjUriBuilder::amount]`.
+///
+/// `ohttp_config` parameter is required only for v2 payjoin.
+/// `[PjUriBuilder::encode_ohttp_config]` and `[PjUriBuilder::decode_ohttp_config]`
+/// can be used to convert `ohttp::KeyConfig` to base64 string and viceversa.
+pub struct PjUriBuilder {
+    /// Address you want to receive funds to.
+    ///
+    /// Must be a valid bitcoin address.
+    address: Address,
+    /// Payjoing endpoint url listening for payjoin requests.
+    ///
+    /// Must be a valid url that can be parsed
+    /// with `[Payjoin::Url::parse]`.
+    pj_endpoint: Url,
+    /// Amount you want to receive.
+    ///
+    /// If `None` the amount will be left unspecified.
+    amount: Option<Amount>,
+    #[cfg(feature = "v2")]
+    /// Config for ohttp.
+    ///
+    /// `[PjUriBuilder::decode_ohttp_config]` can be used to convert
+    /// base64 string to `ohttp::KeyConfig`.
+    ///
+    /// Required only for v2 payjoin.
+    ohttp_config: ohttp::KeyConfig,
+}
+
+impl PjUriBuilder {
+    /// Create a new `PjUriBuilder` with required parameters.
+    pub fn new(
+        address: Address,
+        pj_endpoint: Url,
+        #[cfg(feature = "v2")] ohttp_config: ohttp::KeyConfig,
+    ) -> Self {
+        Self {
+            address,
+            pj_endpoint,
+            amount: None,
+            #[cfg(feature = "v2")]
+            ohttp_config,
+        }
+    }
+
+    /// Set the amount you want to receive.
+    pub fn amount(mut self, amount: Option<bitcoin::Amount>) -> Self {
+        self.amount = amount;
+        self
+    }
+
+    #[cfg(feature = "v2")]
+    /// Convert ohttp::KeyConfig to base64 string
+    pub fn encode_ohttp_config(s: &ohttp::KeyConfig) -> Result<String, PjParseError> {
+        let res = s.encode().map_err(InternalPjParseError::BadOhttp)?;
+        let res = bitcoin::base64::encode_config(res, bitcoin::base64::URL_SAFE);
+        Ok(res)
+    }
+
+    #[cfg(feature = "v2")]
+    /// Convert base64 string to ohttp::KeyConfig
+    pub fn decode_ohttp_config(s: &str) -> Result<ohttp::KeyConfig, PjParseError> {
+        let res = bitcoin::base64::decode_config(s, bitcoin::base64::URL_SAFE)
+            .map_err(InternalPjParseError::NotBase64)?;
+        let res = ohttp::KeyConfig::decode(&res).map_err(InternalPjParseError::BadOhttp)?;
+        Ok(res)
+    }
```

Comment:
These functions don't seem like public `PjUriBuilder` api but rather private `uri` mod functions. They should move outside of this block and be private.